builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -118,10 +119,73 @@ func resourceAwsAutoscalingGroup() *schema.Resource {
 					return hashcode.String(v.(string))
 				},
 			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"propagate": &schema.Schema{
+							Type:     schema.TypeBool,
+							Required: true,
```

Comment:
I wasn't sure what default people would like (I assume not propagated), so I made it mandatory so people had to choose.

I'd be happy to make it optional / default to false if that's what people think is best.